airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -463,15 +463,17 @@ def __init__(
         region_name: str | None = None,
         client_type: str | None = None,
         resource_type: str | None = None,
-        config: Config | None = None,
+        config: Config | dict[str, Any] | None = None,
     ) -> None:
         super().__init__()
         self.aws_conn_id = aws_conn_id
         self.client_type = client_type
         self.resource_type = resource_type
 
         self._region_name = region_name
-        self._config = config
+        if config and isinstance(config, dict):
+            config = Config(**config)
+        self._config = config or None
```

Comment:
A bit dumb check, however it required to pass tests, some of AWS test patch `isinstance` as result `isinstance(None, dict)` evaluate as True